### ISSUE TITLE
Unsafe readObject when "unsealing" CipherText

### ISSUE THREAD

**Issue by h3xstream** [ESAPI/esapi-java-legacy#353]
The method `unseal()` and `verifySignature()` are using indirectly unsafe readObject() with potential user input. This could lead to remote code execution.

[JavaEncryptor.java:827](https://github.com/ESAPI/esapi-java-legacy/blob/20438dbae5ecf8a51f454f38d1e0525fe4b61eb6/src/main/java/org/owasp/esapi/reference/crypto/JavaEncryptor.java#L827)

Path to the unsafe readObject() : 
- [DefaultEncoder.java:446](https://github.com/ESAPI/esapi-java-legacy/blob/20438dbae5ecf8a51f454f38d1e0525fe4b61eb6/src/main/java/org/owasp/esapi/reference/DefaultEncoder.java#L446)
- [Base64.java:1106](https://github.com/ESAPI/esapi-java-legacy/blob/20438dbae5ecf8a51f454f38d1e0525fe4b61eb6/src/main/java/org/owasp/esapi/codecs/Base64.java#L1106)

**Comment by h3xstream** [comment]
My bad .. The `Base64.decodeToObject()` is not use anymore .. This may apply to older version.

**Comment by kwwall** [comment]
Closing issue; nothing to fix here.

I looked way back (at least to 2011/07/13) to the previous commit to DefaultEncoder and even then its decodeFromBase64() method was using Base64.decode() and not the vulnerable Base64.decodeToObject(). In fact, I just grepped all the ESAPI current code base from HEAD as of 2016/01/07 and found no uses of Base64.decodeToObject() and no other uses of readObject other than the one noted in Base64.decodeToObject() [which I was already aware of and thought I had created an issue for; if not, will do so as soon as I close this] nor any calls to readExternal either.
